test/validation/api/event/event.c

```diff
@@ -11,6 +11,197 @@
+	char name[ODP_POOL_NAME_LEN];
+	odp_pool_capability_t capa;
+
+	memset(&capa, 0, sizeof(capa));
+	if (odp_pool_capability(&capa) != 0)
+		return -1;
+
+	ctx->event_types[n] = ODP_EVENT_BUFFER;
+	ctx->pool_types[n++] = ODP_POOL_BUFFER;
+
+	ctx->event_types[n] = ODP_EVENT_PACKET;
+	ctx->pool_types[n++] = ODP_POOL_PACKET;
+
+	ctx->event_types[n] = ODP_EVENT_TIMEOUT;
+	ctx->pool_types[n++] = ODP_POOL_TIMEOUT;
+
+	if (capa.event_vector.max_num > 0) {
+		ctx->event_types[n] = ODP_EVENT_VECTOR;
+		ctx->pool_types[n++] = ODP_POOL_EVENT_VECTOR;
+	}
+
+	if (capa.vector.max_num > 0) {
+		ctx->event_types[n] = ODP_EVENT_PACKET_VECTOR;
+		ctx->pool_types[n++] = ODP_POOL_VECTOR;
+	}
+
+	ctx->num_types = n;
+
+	for (i = 0; i < ctx->num_types; i++) {
+		odp_pool_param_t prm;
+
+		odp_pool_param_init(&prm);
+
+		prm.type = ctx->pool_types[i];
+		switch (ctx->pool_types[i]) {
+		case ODP_POOL_BUFFER:
+			prm.buf.num = NUM_EVENTS;
+			prm.buf.size = EVENT_SIZE;
+			prm.buf.uarea_size = sizeof(uint64_t);
+		break;
+		case ODP_POOL_PACKET:
+			prm.pkt.num = NUM_EVENTS;
+			prm.pkt.len = EVENT_SIZE;
+			prm.pkt.uarea_size = sizeof(uint64_t);
+		break;
+		case ODP_POOL_TIMEOUT:
+			prm.tmo.num = NUM_EVENTS;
+			prm.tmo.uarea_size = sizeof(uint64_t);
+		break;
+		case ODP_POOL_EVENT_VECTOR:
+			prm.event_vector.num = NUM_EVENTS;
+			prm.event_vector.max_size = 1;
+			prm.event_vector.uarea_size = sizeof(uint64_t);
+		break;
+		case ODP_POOL_VECTOR:
+			prm.vector.num = NUM_EVENTS;
+			prm.vector.max_size = 1;
+			prm.vector.uarea_size = sizeof(uint64_t);
+		break;
+		case ODP_POOL_DMA_COMPL:
+		case ODP_POOL_ML_COMPL:
+		break;
+		}
+
+		snprintf(name, ODP_POOL_NAME_LEN, "event_type_compliance_%d", i);
+		ctx->pools[i] = odp_pool_create(name, &prm);
+		if (ctx->pools[i] == ODP_POOL_INVALID)
+			return -1;
+
+		for (j = 0; j < EVENT_BURST; j++) {
+			ctx->events[i][j] = event_types_alloc_event(ctx->pool_types[i]);
+			if (ctx->events[i][j] == ODP_EVENT_INVALID)
+				return -1;
+		}
+	}
+
+	return 0;
+}
+
+static int event_types_suite_term(void)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+	int i;
+
+	for (i = 0; i < ctx->num_types; i++) {
+		odp_event_free_multi(ctx->events[i], EVENT_BURST);
+		(void)odp_pool_destroy(ctx->pools[i]);
+		ctx->pools[i] = ODP_POOL_INVALID;
+	}
+
+	return 0;
+}
+
+static void event_test_pool_info_ctx(void)
+{
+	event_type_ctx_t *ctx = &g_type_ctx;
+	odp_event_t ev;
+	odp_event_type_t exp_type;
+	odp_pool_t ev_pool;
+	odp_pool_info_t info;
+	int i, j;
+
+	for (i = 0; i < ctx->num_types; i++) {
+		for (j = 0; j < EVENT_BURST; j++) {
+			ev = ctx->events[i][j];
+			exp_type = ctx->event_types[i];
+			ev_pool = odp_event_pool(ev);
+
+			if (exp_type == ODP_EVENT_TIMEOUT) {
+				CU_ASSERT_FATAL(ev_pool == ODP_POOL_INVALID);
+			} else {
+				CU_ASSERT_FATAL(ev_pool != ODP_POOL_INVALID);
+				CU_ASSERT_FATAL(odp_pool_info(ev_pool, &info) == 0);
+				CU_ASSERT(info.type == ctx->pool_types[i]);
+				CU_ASSERT(odp_event_pool(ev) == ev_pool);
```

Comment:
Could be simplified to `CU_ASSERT(ev_pool == ctx->pool_types[i]);` and function name changed to `event_test_pool`.